cpp/scenario/gazebo/include/scenario/gazebo/Model.h

```diff
@@ -0,0 +1,222 @@
+                             const std::vector<std::string>& jointNames = {});
+
+    std::vector<LinkPtr> links( //
+        const std::vector<std::string>& linkNames = {}) const;
+    std::vector<JointPtr> joints( //
+        const std::vector<std::string>& jointNames = {}) const;
+
+    // =========================
+    // Vectorized Target Methods
+    // =========================
+
+    bool setJointForces( //
+        const std::vector<double> forces,
+        const std::vector<std::string>& jointNames = {});
+    bool setJointPositionTargets( //
+        const std::vector<double> positions,
+        const std::vector<std::string>& jointNames = {});
+    bool setJointVelocityTargets( //
+        const std::vector<double> velocities,
+        const std::vector<std::string>& jointNames = {});
+    bool setJointAccelerationTargets( //
+        const std::vector<double> velocities,
+        const std::vector<std::string>& jointNames = {});
+
+    bool resetJointPositions( //
+        const std::vector<double> positions,
+        const std::vector<std::string>& jointNames = {});
+    bool resetJointVelocities( //
+        const std::vector<double> velocities,
+        const std::vector<std::string>& jointNames = {});
+
+    std::vector<double> jointPositionTargets( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointVelocityTargets( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointAccelerationTargets( //
+        const std::vector<std::string>& jointNames = {}) const;
+
+    // =========
+    // Base Link
+    // =========
+
+    std::string baseFrame() const;
+    bool setBaseFrame(const std::string& frameName);
+
+    bool fixedBase() const;
+    bool setAsFixedBase(const bool fixedBase);
+
+    std::array<double, 6> baseWrench() const;
+    std::array<double, 3> basePosition() const;
+    std::array<double, 4> baseOrientation() const;
+    std::array<double, 3> baseLinearVelocity() const;
+    std::array<double, 3> baseAngularVelocity() const;
+
+    bool resetBaseLinearVelocity(const std::array<double, 3>& linear);
+    bool resetBaseAngularVelocity(const std::array<double, 3>& angular);
+    bool resetBaseVelocity(const std::array<double, 3>& linear,
+                           const std::array<double, 3>& angular);
+
+    bool resetBasePose(const std::array<double, 3>& position,
+                       const std::array<double, 4>& orientation);
+    bool resetBasePosition(const std::array<double, 3>& position);
+    bool resetBaseOrientation(const std::array<double, 4>& orientation);
+
+    // =================
+    // Base Link Targets
+    // =================
+
+    bool setBasePoseTarget(const std::array<double, 3>& position,
+                           const std::array<double, 4>& orientation);
+    bool setBasePositionTarget(const std::array<double, 3>& position);
+    bool setBasOrientationTarget(const std::array<double, 4>& orientation);
```

Comment:
```suggestion
    bool setBaseOrientationTarget(const std::array<double, 4>& orientation);
```

Reply:
Woops, excellent catch!!

Reply:
Addressed in https://github.com/robotology/gym-ignition/pull/158/commits/198185b918c3be935f7c9af47ec67213b7939dd9